cmd/notation/inspect.go

```diff
@@ -0,0 +1,288 @@
+		for _, sigManifestDesc := range signatureManifests {
+			sigBlob, sigDesc, err := sigRepo.FetchSignatureBlob(ctx, sigManifestDesc)
+			if err != nil {
+				return fmt.Errorf("unable to retrieve digital signature with digest %q associated with %q from the registry, error : %v", sigManifestDesc.Digest, manifestDesc.Digest, err.Error())
+			}
+
+			envelope, err := signature.ParseEnvelope(sigDesc.MediaType, sigBlob)
+			envelopeContent, err := envelope.Content()
+			if err != nil {
+				return err
+			}
+
+			signedArtifactDesc, err := notation.GetDescriptorFromPayload(&envelopeContent.Payload)
+			if err != nil {
+				return err
+			}
+
+			sig := signatureOutput{
+				Digest:                sigDesc.Digest.String(),
+				SignatureAlgorithm:    envelopeContent.SignerInfo.SignatureAlgorithm.String(),
+				SignedAttributes:      getSignedAttributes(envelopeContent),
+				UserDefinedAttributes: signedArtifactDesc.Annotations,
+				UnsignedAttributes:    getUnsignedAttributes(envelopeContent),
+				Certificates:          getCertificates(envelopeContent),
+				SignedArtifact: artifact{
+					MediaType: signedArtifactDesc.MediaType,
+					Digest:    signedArtifactDesc.Digest.String(),
+					Size:      signedArtifactDesc.Size,
+				},
+			}
+
+			output.Signatures = append(output.Signatures, sig)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return printOutput(opts.outputFormat, output)
+}
+
+func getSignedAttributes(envContent *signature.EnvelopeContent) map[string]string {
+	signedAttributes := map[string]string{}
+
+	signedAttributes["signingScheme"] = string(envContent.SignerInfo.SignedAttributes.SigningScheme)
+	signedAttributes["signingTime"] = envContent.SignerInfo.SignedAttributes.SigningTime.String()
+	signedAttributes["expiry"] = envContent.SignerInfo.SignedAttributes.Expiry.String()
+
+	for _, attribute := range envContent.SignerInfo.SignedAttributes.ExtendedAttributes {
+		if key, ok := attribute.Key.(string); ok {
+			if value, ok := attribute.Value.(string); ok {
+				signedAttributes[key] = value
+			}
+		}
+	}
+
+	return signedAttributes
+}
+
+func getUnsignedAttributes(envContent *signature.EnvelopeContent) map[string]string {
+	unsignedAttributes := map[string]string{}
+
+	if envContent.SignerInfo.UnsignedAttributes.TimestampSignature != nil {
+		unsignedAttributes["timestampSignature"] = string(envContent.SignerInfo.UnsignedAttributes.TimestampSignature)
+	}
+
+	if envContent.SignerInfo.UnsignedAttributes.SigningAgent != "" {
+		unsignedAttributes["signingAgent"] = envContent.SignerInfo.UnsignedAttributes.SigningAgent
+	}
+
+	return unsignedAttributes
+}
+
+func getCertificates(envContent *signature.EnvelopeContent) []certificateOutput {
+	certificates := []certificateOutput{}
+
+	for _, cert := range envContent.SignerInfo.CertificateChain {
+		h := sha1.New()
+		h.Write(cert.RawTBSCertificate)
+		fingerprint := fmt.Sprintf("%x", h.Sum(nil))
```

Comment:
Should we use `cert.Raw` to compute SHA1 Thumbprint?
```suggestion
		h := sha1.Sum(cert.Raw)
		fingerprint := strings.ToLower(hex.EncodeToString(h[:]))
```